applications/GeoMechanicsApplication/custom_strategies/schemes/newmark_quasistatic_U_Pw_scheme.hpp

```diff
@@ -17,167 +17,135 @@
-
-            KRATOS_ERROR_IF_NOT(rNode.HasDofFor(DISPLACEMENT_X))
-                << "missing DISPLACEMENT_X dof on node "
-                << rNode.Id()
-                << std::endl;
-
-            KRATOS_ERROR_IF_NOT(rNode.HasDofFor(DISPLACEMENT_Y))
-                << "missing DISPLACEMENT_Y dof on node "
-                << rNode.Id()
-                << std::endl;
-
-            KRATOS_ERROR_IF_NOT(rNode.HasDofFor(DISPLACEMENT_Z))
-                << "missing DISPLACEMENT_Z dof on node "
-                << rNode.Id()
-                << std::endl;
-
-            KRATOS_ERROR_IF_NOT(rNode.HasDofFor(WATER_PRESSURE))
-                << "missing WATER_PRESSURE dof on node "
-                << rNode.Id()
-                << std::endl;
+        // check that variables are correctly allocated
+        for (const auto& r_node : rModelPart.Nodes()) {
+            this->CheckSolutionStepsData(r_node, DISPLACEMENT);
+            this->CheckSolutionStepsData(r_node, VELOCITY);
+            this->CheckSolutionStepsData(r_node, ACCELERATION);
+            this->CheckSolutionStepsData(r_node, WATER_PRESSURE);
+            this->CheckSolutionStepsData(r_node, DT_WATER_PRESSURE);
+
+            this->CheckDof(r_node, DISPLACEMENT_X);
+            this->CheckDof(r_node, DISPLACEMENT_Y);
+            this->CheckDof(r_node, DISPLACEMENT_Z);
+            this->CheckDof(r_node, WATER_PRESSURE);
         }
 
         this->CheckBufferSize(rModelPart);
 
         // Check beta, gamma and theta
-        KRATOS_ERROR_IF(mBeta <= 0.0 || mGamma<= 0.0 || mTheta <= 0.0)
-            << "Some of the scheme variables: beta, gamma or theta has an invalid value"
-            << std::endl;
+        KRATOS_ERROR_IF(mBeta <= 0.0 || mGamma <= 0.0) << "Some of the scheme variables: beta or gamma has an invalid value"
+                                                                              << std::endl;
 
         return 0;
 
-        KRATOS_CATCH( "" )
+        KRATOS_CATCH("")
     }
 
-    void FinalizeSolutionStep( ModelPart& rModelPart,
-                               TSystemMatrixType& A,
-                               TSystemVectorType& Dx,
-                               TSystemVectorType& b) override
+    void FinalizeSolutionStep(ModelPart& rModelPart,
+                              TSystemMatrixType& A,
+                              TSystemVectorType& Dx,
+                              TSystemVectorType& b) override
     {
         KRATOS_TRY
 
         if (rModelPart.GetProcessInfo()[NODAL_SMOOTHING]) {
-            unsigned int Dim = rModelPart.GetProcessInfo()[DOMAIN_SIZE];
+            unsigned int dim = rModelPart.GetProcessInfo()[DOMAIN_SIZE];
 
-            SizeType StressTensorSize = STRESS_TENSOR_SIZE_2D;
-            if (Dim == N_DIM_3D) StressTensorSize = STRESS_TENSOR_SIZE_3D; 
+            SizeType stress_tensor_size = STRESS_TENSOR_SIZE_2D;
+            if (dim == N_DIM_3D)
+                stress_tensor_size = STRESS_TENSOR_SIZE_3D;
```

Comment:
stress_tensor_size = ( dim == N_DIM_3D ? STRESS_TENSOR_SIZE_3D : STRESS_TENSOR_SIZE_2D );
not sure about the correct syntax, but I would expect STRESS_TENSOR_SIZE_PLANE_STRAIN ( maybe that does not exist, but at least the Voigt vector size differs between plane stress, plane strain and 3D ( 3, 4 and 6 as values ).